client/state/sites/launch/actions.js

```diff
@@ -28,17 +35,44 @@ export const launchSiteFailure = ( siteId ) => ( {
 	siteId,
 } );
 
+export const launchSiteSuccessCelebration = ( siteId ) => ( {
+	type: SITE_LAUNCH_SUCCESS_CELEBRATION,
+	siteId,
+} );
+
+export const removeSiteLaunchCelebration = ( siteId ) => ( {
+	type: SITE_LAUNCH_SUCCESS_CELEBRATION_REMOVE,
+	siteId,
+} );
+
 /**
  * @param {number} siteId
  * @param {string?} source
  */
 export const launchSiteOrRedirectToLaunchSignupFlow =
 	( siteId, source = null, siteTitle = null, search = null ) =>
-	( dispatch, getState ) => {
+	async ( dispatch, getState ) => {
 		if ( ! isUnlaunchedSite( getState(), siteId ) ) {
 			return;
 		}
 
+		const experiment = await getLaunchExperimentAssignment();
```

Comment:
Here's the experiment logic, centralized in one action. We decided the user's path based on the source of the call.